src/EPR.SubsidiaryBulkUpload.Application.UnitTests/Services/NotificationServiceTests.cs

```diff
@@ -100,4 +101,148 @@ public async Task SetErrorStatus_ShouldSerializeErrorsAndSetStringInRedisAndLogI
+
+        var json = JsonSerializer.Serialize(previousErrors);
+        _redisDatabaseMock.Setup(db => db.StringGetAsync(It.IsAny<RedisKey>(), It.IsAny<CommandFlags>()))
+            .ReturnsAsync((RedisValue)json);
+
+        previousErrors.Errors.AddRange(errorsModel);
+        var serializedErrors = JsonSerializer.Serialize(new { Errors = previousErrors.Errors });
+
+        // Act
+        await _notificationService.SetErrorStatus(key, errorsModel);
+
+        // Assert
+        _redisDatabaseMock.Verify(db => db.StringSetAsync(key, serializedErrors, null, false, When.Always, CommandFlags.None), Times.Once);
+        _loggerMock.VerifyLog(x => x.LogInformation("Redis updated key: {Key} errors: {Value}", key, serializedErrors), Times.Once);
+    }
+
+    [TestMethod]
+    public async Task GetNotificationErrorsAsync_EmptyKey_ReturnsEmptyResponse()
+    {
+        // Arrange
+        var key = "emptyKey";
+
+        _redisDatabaseMock.Setup(db => db.StringGetAsync(It.IsAny<RedisKey>(), It.IsAny<CommandFlags>()))
+            .ReturnsAsync(RedisValue.Null);
+
+        // Act
+        var result = await _notificationService.GetNotificationErrorsAsync(key);
+
+        // Assert
+        Assert.IsNotNull(result);
+        Assert.AreEqual(result.Errors.Count, 0);
+    }
+
+    [TestMethod]
+    public async Task GetNotificationErrorsAsync_ValidKey_ReturnsErrorsResponse()
+    {
+        // Arrange
+        var key = "validKey";
+        var errors = new UploadFileErrorResponse
+        {
+            Errors = new List<UploadFileErrorModel>
+                {
+                    new UploadFileErrorModel { FileLineNumber = 1, FileContent = "Content1", Message = "Message1", IsError = true, ErrorNumber = 6 },
+                    new UploadFileErrorModel { FileLineNumber = 2, FileContent = "Content2", Message = "Message2", IsError = false, ErrorNumber = 9 }
+                }
+        };
+
+        var json = JsonSerializer.Serialize(errors);
+        _redisDatabaseMock.Setup(db => db.StringGetAsync(It.IsAny<RedisKey>(), It.IsAny<CommandFlags>()))
+            .ReturnsAsync((RedisValue)json);
+
+        // Act
+        var result = await _notificationService.GetNotificationErrorsAsync(key);
+
+        // Assert
+        Assert.IsNotNull(result);
+        Assert.AreEqual(2, result.Errors.Count);
+        Assert.AreEqual(1, result.Errors[0].FileLineNumber);
+        Assert.AreEqual("Content1", result.Errors[0].FileContent);
+        Assert.AreEqual("Message1", result.Errors[0].Message);
+        Assert.AreEqual(6, result.Errors[0].ErrorNumber);
+        Assert.IsTrue(result.Errors[0].IsError);
+        Assert.AreEqual(2, result.Errors[1].FileLineNumber);
+        Assert.AreEqual("Content2", result.Errors[1].FileContent);
+        Assert.AreEqual("Message2", result.Errors[1].Message);
+        Assert.IsFalse(result.Errors[1].IsError);
+        Assert.AreEqual(9, result.Errors[1].ErrorNumber);
+
+        _loggerMock.Verify(
```

Comment:
This can be simplified to
`_loggerMock.VerifyLog(x => x.LogInformation("Redis errors response key: {Key} errors: {Value}", key, json), Times.Once);`

Reply:
changed